File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n = nums.size();
        unordered_map<int, int> idx;
        
        for(int i = 0; i < n; i++){
            int toHave = target - nums[i];
            if(idx[toHave] > 0)
                return {i, idx[toHave]-1};
            idx[nums[i]] = i + 1;
        }
        return {-1, -1};
    }
};

File: 1-two-sum/1-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int n = nums.Length;
        Dictionary<int, int> dict = new Dictionary<int, int>();
        
        
        for(int i = 0; i < n; i++){
            int needed = target - nums[i];
            if(dict.ContainsKey(needed)){
                int[] res = {dict[needed], i};
                return res;
            }
            if(!dict.ContainsKey(nums[i]))
                dict.Add(nums[i], i);    
        }
        
        return nums;
    }
}